app/src/handle-request.lisp

```diff
@@ -166,4 +149,38 @@ The mapping vector V specifies that the qubit as specified in the program V[i] h
                  (qvm:map-amplitudes
                   qvm
                   (lambda (z) (write-complex-double-float-as-binary z reply-stream)))))
-             (format-log "Response sent in ~D ms." send-response-time))))))))
+             (format-log "Response sent in ~D ms." send-response-time))))
+        
+        ((:probabilities)
+         (check-for-quil-instrs-field js)
+         (let* ((isns (get-quil-instrs-field js))
+                (quil (let ((quil:*allow-unresolved-applications* t))
+                        (process-quil (safely-parse-quil-string isns))))
+                (num-qubits (cl-quil:qubits-needed quil)))
+           (let (send-response-time)
+             (multiple-value-bind (qvm probabilities)
```

Comment:
I feel sad that we cons up probabilities here, only to fire them off below. But we will have to do that for RPCQ currently, so it's probably not unreasonable to do... 😞

Reply:
😿